src/main/java/com/thealgorithms/sorts/CountingSort.java

```diff
@@ -1,93 +1,53 @@
 package com.thealgorithms.sorts;
 
-import static com.thealgorithms.sorts.SortUtils.print;
-import static java.util.stream.Collectors.toList;
-import static java.util.stream.Collectors.toMap;
-
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 import java.util.TreeMap;
-import java.util.stream.IntStream;
-import java.util.stream.Stream;
 
 /**
+ * CountingSort is a generic implementation of the counting sort algorithm.
+ * This implementation sorts elements that implement the Comparable interface.
+ *
  * @author Youssef Ali (https://github.com/youssefAli11997)
  * @author Podshivalov Nikita (https://github.com/nikitap492)
  */
 class CountingSort implements SortAlgorithm {
-
     @Override
-    public <T extends Comparable<T>> T[] sort(T[] unsorted) {
-        return sort(Arrays.asList(unsorted)).toArray(unsorted);
+    public <T extends Comparable<T>> T[] sort(T[] array) {
+        return sort(Arrays.asList(array)).toArray(array);
     }
 
     /**
-     * This method implements the Generic Counting Sort
+     * Sorts the provided list using the counting sort algorithm.
      *
-     * @param list The list to be sorted
-     * <p>
-     * Sorts the list in increasing order The method uses list elements as keys
-     * in the frequency map
+     * @param list The list to be sorted.
+     * @param <T> The type of elements in the list, must be Comparable.
+     * @return A sorted list in increasing order.
      */
     @Override
     public <T extends Comparable<T>> List<T> sort(List<T> list) {
-        Map<T, Integer> frequency = new TreeMap<>();
-        // The final output array
-        List<T> sortedArray = new ArrayList<>(list.size());
-
-        // Counting the frequency of @param array elements
-        list.forEach(v -> frequency.put(v, frequency.getOrDefault(v, 0) + 1));
+        Map<T, Integer> frequencyMap = computeFequencyMap(list);
+        return extractSortedArray(frequencyMap);
```

Comment:
In order to stop all of the discussions about the type, name and if it should be `final`, I would simply
```suggestion
        return extractSortedArray(computeFequencyMap(list));
```